wcfsetup/install/files/lib/data/user/UserAction.class.php

```diff
@@ -828,4 +828,47 @@ public function validateSaveSocialNetworkPrivacySettings() {
 	public function saveSocialNetworkPrivacySettings() {
 		// does nothing
 	}
+	
+	/**
+	 * Validates the 'resendActivationMail' action.
+	 * @throws UserInputException
+	 * @since 3.2
+	 */
+	public function validateResendActivationMail() {
+		$this->readObjects();
+		
+		foreach ($this->objects as $object) {
+			if (!$object->activationCode) {
+				throw new UserInputException('objectIDs');
+			}
+		}
+	}
+	
+	/**
+	 * Triggers a new activation email.
+	 * @since 3.2
+	 */
+	public function resendActivationMail() {
+		foreach ($this->objects as $object) {
+			$code = UserRegistrationUtil::getActivationCode();
+			$action = new UserAction($this->objects, 'update', [
+				'data' => [
+					'activationCode' => $code
+				]
+			]);
+			$action->executeAction();
+			
+			$email = new Email();
+			// get current/updated user object for activation code
+			$email->addRecipient(new UserMailbox(new User($object->userID)));
```

Comment:
Create a `UserList` with all the users you need updated (saves queries). This is most easily done using a two-step process:

1. Update the activationCode for every user.
2. Select all the users using a `UserList`.
3. Send email as usual.